.github/workflows/ci.yml

```diff
@@ -0,0 +1,106 @@
+name: Rust CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+  schedule:
+    - cron: '11 7 * * 1,4'
+
+# env:
+#   RUSTFLAGS: -Dwarnings
+#   RUSTDOCFLAGS: -Dwarnings
+
+jobs:
+
+  fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          components: rustfmt
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
+
+  # clippy:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v1
+  #     - uses: actions-rs/toolchain@v1
+  #       with:
+  #           toolchain: stable
+  #           profile: minimal
+  #           components: clippy
+  #           override: true
+  #     - uses: actions-rs/clippy-check@v1
+  #       with:
+  #         token: ${{ secrets.GITHUB_TOKEN }}
+  #         args: --workspace --tests --examples
+
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+          components: rust-docs
+          override: true
+      - uses: swatinem/rust-cache@v1
+        with:
+          key: ${{ github.job }}
+      - uses: actions-rs/cargo@v1
+        with:
+          command: doc
+          args: --workspace --no-deps
+
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        rust: [nightly, stable]
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          profile: minimal
+          override: true
+      - uses: swatinem/rust-cache@v1
+        with:
+          key: ${{ github.job }}
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --workspace --tests --examples --bins
+  
+  doctest:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        rust: [nightly, beta, stable]
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          profile: minimal
+          override: true
+      - uses: swatinem/rust-cache@v1
+        with:
+          key: ${{ github.job }}
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --workspace --doc
```

Comment:
Is this split out to a separate job because that's how it currently is in the Travis config? I'm fuzzy on the details but looking at it I think it's only written that way because the `docmatic` crate had some problems, and I believe I'm only using that to test the examples in the `README`:

What would you think about swapping that out for the `doc-comment` crate (it's trivial to use), running `cargo test --all` in the `test` job above, and removing this separate doctest job?
https://docs.rs/doc-comment/0.3.3/doc_comment/

Reply:
I would prefer one job as well, but I couldn't figure out the flags that run everything at once.  It seems there are some tests that this runs which aren't run in the main test job.  I don't know the doc-comment crate, I'll check it out and see whether I can get rid of this step

Reply:
For most of my crates I think I just run `cargo test --all` and it runs everything I've ever cared about? I'm not sure if there's something covered by `cargo doc` that doesn't get covered there.